### PR TITLE
Custom username

### PR DESCRIPTION
**Pull request by iheron**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,3 +26,14 @@ client, err := NewMultiClient(account, "identifier", 3, true, conf)
 client.Send(nkn.NewStringArray("DNS:foo.com"), "Hello world.", nil)
 ```
 
+## Use custom username
+
+Address in a dns TXT record use custom username, like this:
+
+> user1.foo.com TXT nkn=abc...def \
```
Comment (yilunzhang) [inline README.md]: Should we use `_nkn.user1.foo.com`?

Comment (yilunzhang) [reply README.md]: I think we should use something like
```
_nkn.foo.com TXT user1@nkn=123abc
_nkn.foo.com TXT user2@nkn=456def
```
This way we can also support username + subdomain like `user@bar.foo.com`

File: utils.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package dnsresolver
+
+import "regexp"
+
+func IsEmail(email string) bool {
+	pattern := `\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*`
+	reg := regexp.MustCompile(pattern)
```
Comment (yilunzhang) [inline utils.go]: Put this as global variable to avoid compile at each function call

File: README.md (hunk 1 above)
Comment (iheron) [reply README.md]: > I think we should use something like
> 
> ```
> _nkn.foo.com TXT user1@nkn=123abc
> _nkn.foo.com TXT user2@nkn=456def
> ```
> 
> This way we can also support username + subdomain like `user@bar.foo.com`

This way maybe only support 10 record. How about `_nkn.user1.foo.com `?

File: README.md [hunk 3]
```diff
@@ -23,3 +26,16 @@ client, err := NewMultiClient(account, "identifier", 3, true, conf)
 client.Send(nkn.NewStringArray("DNS:foo.com"), "Hello world.", nil)
 ```
 
+## Use custom username
+
+Address in a dns TXT record use custom username, like this:
+
+* Username do not include "."
```
Comment (yilunzhang) [inline README.md]: We no longer has this constraint right?

**Review by yilunzhang — APPROVED** [review]

**Pull request merged**